content/PHPDataObjects/DataObjects.q.md

```diff
@@ -0,0 +1,1145 @@
+
+    try {
+        $conn = new PDO($dsn);
+        if ($conn) {
+            echo "Successfully connected to the database. ";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection."; 
+    }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once ("connection.php");
+
+    $stmt = $conn->query("SELECT * FROM students");
+    foreach ($stmt as $row) {
+        echo $row['student_id'].$row['firstname'].$row['lastname'].$row['birthdate'].$row['gender']. ' ';
+    }
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.` and `JohnSmith1999-10-02Male`.
+
+- It prints `Successfully connected to the database.` and `1JohnSmith1999-02-10Male`.
+
+
+/// type=MS, answer=[3,5]
+
+Which statements correctly describe the `foreach` statement on line 5 of `student.php`?
+
+- It removes all the elements in `$stmt`.
+
+- It accesses all the elements in `$stmt`.
+
+- It loops over the `$stmt` object as the array expression.
+
+- It removes all the elements in the multidimensional array `$stmt`.
+
+- It assigns the value of the current element on each iteration to `$row`.
+
+
+/// type=SS, answer=[2]
+
+What value is assigned to `student_id`?
+
+- `0`
+
+- `1`
+
+- `John`
+
+- `Smith`
+
+- No value is assigned.
+
+
+/// type=MS, answer=[3,4,5]
+
+Which statements correctly describe the statement `$stmt = $conn->query("SELECT * FROM students");` on line 4 of `student.php`?
+
+- It assigns the value of `$conn` to `$stmt`.
+
+- It ends the execution of the `query()` method. 
+
+- It calls the `query()` method of the `$conn` object.
+
+- It assigns the result set `PDOStatement` object to the variable `$stmt`.
+
+- It executes the argument `SELECT * FROM students` in the `query()` method.
+
+
+:::
+
+
++++
+
+
++++
+
+### Part 2: Knowledge Assessment
+
+/// type=SS, answer=[5]
+
+Which statement best describes `PDO::query()`?
+
+- It sends queries or uploads data to certain databases.
+
+- It defines an interface in PHP to access certain databases.
+
+- It refers to the server’s hostname of the PostgreSQL database.
+
+- It represents a connection between a database server and PHP.
+
+- It executes an SQL statement and returns a `PDOStatement` object.
+
+
+/// type=SS, answer=[2]
+
+What parameter is passed in the `PDO::query()` method?
+
+- A DSN parameter.
+
+- An SQL statement.
+
+- An `echo` statement.
+
+- A `PDOStatement` object.
+
+- A `PDO_PGQSL DSN` statement.
+
+
+/// type=SS, answer=[5]
+
+What value does the `PDO::query()` method return?
+
+- A DSN parameter.
+
+- An SQL statement.
+
+- A `PDO_PGSQL DSN` statement.
+
+- A `true` or `false` boolean value.
+
+- A `PDOStatement` object or `false` on failure.
+
+
+/// type=MS, answer=[3,4,5]
+
+Which statements best describe a `PDOStatement`?
```

Comment:
Grammar error here....

Please replace `best` with `correctly`. It is not correct to use `best` when you have multiple answers. Please check. Thanks.